lib/puma/server.rb

```diff
@@ -541,10 +541,17 @@ def lowlevel_error(e, env, status=500)
         backtrace = e.backtrace.nil? ? '<no backtrace available>' : e.backtrace.join("\n")
         [status, {}, ["Puma caught this error: #{e.message} (#{e.class})\n#{backtrace}"]]
       else
-        [status, {}, ["An unhandled lowlevel error occurred. The application logs may have details.\n"]]
+        [status, {}, ["An error occurred\n"]]
```

Comment:
I'm not sure this is that much better than the previous message, it is probably still somewhat unique to Puma.

I checked what [pitchfork](https://github.com/Shopify/pitchfork) does in [the `500 Internal Server Error` case](https://github.com/Shopify/pitchfork/blob/b5e2bbcbebd22bcd8a212c300e0bc8abbecc4040/test/integration/test_broken_app.rb), and it doesn't respond with any body (`RACK_ENV=development` and `RACK_ENV=production` behaves the same).

```
$ curl -s -v http://127.0.0.1:8080/raise
*   Trying 127.0.0.1:8080...
* Connected to 127.0.0.1 (127.0.0.1) port 8080 (#0)
> GET /raise HTTP/1.1
> Host: 127.0.0.1:8080
> User-Agent: curl/7.84.0
> Accept: */*
>
* Mark bundle as not supporting multiuse
< HTTP/1.1 500 Internal Server Error
* no chunk, no close, no size. Assume close to signal end
<
* Closing connection 0

$ curl -s -v http://127.0.0.1:8080/nil
*   Trying 127.0.0.1:8080...
* Connected to 127.0.0.1 (127.0.0.1) port 8080 (#0)
> GET /nil HTTP/1.1
> Host: 127.0.0.1:8080
> User-Agent: curl/7.84.0
> Accept: */*
>
* Mark bundle as not supporting multiuse
< HTTP/1.1 500 Internal Server Error
* no chunk, no close, no size. Assume close to signal end
<
* Closing connection 0
```